core/domain/rating_services_test.py

```diff
@@ -150,52 +158,67 @@ def test_rating_assignations_do_not_conflict(self):
+    def test_invalid_ratings_are_forbidden(self) -> None:
+        with self.assertRaisesRegex(  # type: ignore[no-untyped-call]
             ValueError, 'Expected a rating 1-5, received 0'
             ):
             rating_services.assign_rating_to_exploration(
                 self.USER_ID_1, self.EXP_ID, 0)
 
-        with self.assertRaisesRegex(
+        with self.assertRaisesRegex(  # type: ignore[no-untyped-call]
             ValueError, 'Expected a rating 1-5, received 7'
             ):
             rating_services.assign_rating_to_exploration(
                 self.USER_ID_1, self.EXP_ID, 7)
 
-        with self.assertRaisesRegex(
+        with self.assertRaisesRegex(  # type: ignore[no-untyped-call]
             ValueError, 'Expected the rating to be an integer, received 2'
             ):
+            # The argument new_rating of assign_rating_to_exploration() can
+            # only accept integer values, but for testing purposes here we are
+            # providing string value. Thus to silent MyPy error, we added an
+            # ignore statement here.
             rating_services.assign_rating_to_exploration(
-                self.USER_ID_1, self.EXP_ID, '2')
+                self.USER_ID_1, self.EXP_ID, '2')  # type: ignore[arg-type]
 
-        with self.assertRaisesRegex(
+        with self.assertRaisesRegex(  # type: ignore[no-untyped-call]
             ValueError, 'Expected the rating to be an integer, received aaa'
             ):
+            # The argument new_rating of assign_rating_to_exploration() can
+            # only accept integer values, but for testing purposes here we are
+            # providing string value. Thus to silent MyPy error, we added an
+            # ignore statement here.
             rating_services.assign_rating_to_exploration(
-                self.USER_ID_1, self.EXP_ID, 'aaa')
+                self.USER_ID_1, self.EXP_ID, 'aaa')  # type: ignore[arg-type]
 
-    def test_invalid_exploration_ids_are_forbidden(self):
-        with self.assertRaisesRegex(
+    def test_invalid_exploration_ids_are_forbidden(self) -> None:
+        with self.assertRaisesRegex(  # type: ignore[no-untyped-call]
             Exception, 'Invalid exploration id invalid_id'
             ):
             rating_services.assign_rating_to_exploration(
                 self.USER_ID_1, 'invalid_id', 3)
 
-    def test_rating_assignation_with_no_exploration_summary_ratings(self):
+    def test_rating_assignation_with_no_exploration_summary_ratings(
+        self
+    ) -> None:
 
-        def _mock_get_exploration_summary_by_id(exp_id):
+        def _mock_get_exploration_summary_by_id(
+            exp_id: str
+        ) -> exp_models.ExpSummaryModel:
             """Assign None to exploration summary ratings."""
             exp_summary_model = exp_models.ExpSummaryModel.get(exp_id)
+
+            # Ruling out the possibility of None for mypy type checking.
+            assert exp_summary_model is not None
```

Comment:
Can you rather use `strict=True` in the `get` call above?

Reply:
I think the default value of `strict` is `True`.
https://github.com/oppia/oppia/blob/cb3dff60e88a796577d7ef75a1ec35d33c6128c0/core/storage/base_model/gae_models.py#L267

And also looking at the function signature i think it always return `Optional[]` whether we use `strict=true` or not, Please correct me if I'm wrong.

Reply:
You're right, can we improve that signature? As we know that when True it cannot be `None`?

Reply:
Done!